### PR TITLE
Adding retrieving API by ID in DevportalAPI

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Purpose
Fix for https://github.com/wso2/apk/issues/241

### CODE REVIEW

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIConsumerImpl.java [hunk 1]
```diff
@@ -2720,7 +2720,14 @@ public ApiTypeWrapper getAPIorAPIProductByUUID(String uuid, String organization)
                     apiProduct = addTiersToAPI(apiProduct, organization);
                     return new ApiTypeWrapper(apiProduct);
                 } else {
-                    API api = APIMapper.INSTANCE.toApi(devPortalApi);
+                    //API api = APIMapper.INSTANCE.toApi(devPortalApi);
```
Comment (AmaliMatharaarachchi) [inline common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIConsumerImpl.java:2723]: Did you keep this comment intentionally?

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/AbstractAPIManager.java [hunk 2]
```diff
@@ -1172,9 +1172,10 @@ protected void populateDevPortalAPIInformation(String uuid, String organization,
         }
         api.setUriTemplates(uriTemplates);
         //CORS . if null is returned, set default config from the configuration
-        if (api.getCorsConfiguration() == null) {
-            api.setCorsConfiguration(APIUtil.getDefaultCorsConfiguration());
-        }
+        //Todo Need to implement for APK
```
Comment (AmaliMatharaarachchi) [inline common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/AbstractAPIManager.java:1175]: shall we follow this practice to keep track of todo items in apk repo? 
```suggestion
        // TODO (CrowleyRajapakse) Need to implement for APK
```

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApiDAOImpl.java [hunk 3]
```diff
@@ -3103,26 +3102,34 @@ public DevPortalAPI getDevPortalAPI(Organization organization, String apiId) thr
...
-                        api.setSwaggerDefinition(APIMgtDBUtil.getStringFromInputStream(apiDefinitionBlob));
-                    } else if (StringUtils.equals("asyncapi.json", mediaType)) {
-                        api.setAsyncApiDefinition(APIMgtDBUtil.getStringFromInputStream(apiDefinitionBlob));
+                        devPortalAPI.setSwaggerDefinition(APIMgtDBUtil.getStringFromInputStream(apiDefinitionBlob));
                     }
-                }
-                API apiObject = APIMapper.INSTANCE.toApi(api);
-                devPortalAPI = APIMapper.INSTANCE.toDevPortalApi(apiObject);
-                if (APIConstants.API_GLOBAL_VISIBILITY.equals(api.getVisibility())) {
+//                    else if (StringUtils.equals("asyncapi.json", mediaType)) {
```
Comment (AmaliMatharaarachchi) [inline common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApiDAOImpl.java:3112]: If you are keeping these commented lines for future use, lets add todo tag to keep track of them. Or remove them completely if not needed anymore

Comment (AmaliMatharaarachchi) [reply common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApiDAOImpl.java:3112]: please check on other commented places too on this PR

File: devportal/devportal-domain-service/java/src/main/java/org/wso2/apk/apimgt/devportal/impl/ApisCommonImpl.java [hunk 4]
```diff
@@ -695,7 +695,33 @@ public static List<Tier> getSubscriptionPolicies(String apiId, String organizati
...
+        ApiTypeWrapper api = apiConsumer.getAPIorAPIProductByUUID(apiId, organization);
+        String status = api.getStatus();
+
+        // Extracting clicked API name by the user, for the recommendation system
+        String userName = RestApiCommonUtil.getLoggedInUsername();
+        apiConsumer.publishClickedAPI(api, userName, organization);
+
+        if (APIConstants.PUBLISHED.equals(status) || APIConstants.PROTOTYPED.equals(status)
+                || APIConstants.DEPRECATED.equals(status)) {
+
```
Comment (AmaliMatharaarachchi) [inline devportal/devportal-domain-service/java/src/main/java/org/wso2/apk/apimgt/devportal/impl/ApisCommonImpl.java:709]:
```suggestion
```

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIConsumerImpl.java (hunk 1 above)
Comment (CrowleyRajapakse) [reply common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIConsumerImpl.java:2724]: Yes, added a TODO task

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/AbstractAPIManager.java (hunk 2 above)
Comment (CrowleyRajapakse) [reply common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/AbstractAPIManager.java]: removed the cors configuration setting

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApiDAOImpl.java (hunk 3 above)
Comment (CrowleyRajapakse) [reply common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApiDAOImpl.java:3113]: added a todo and resolved

File: devportal/devportal-domain-service/java/src/main/java/org/wso2/apk/apimgt/devportal/impl/ApisCommonImpl.java (hunk 4 above)
Comment (CrowleyRajapakse) [reply devportal/devportal-domain-service/java/src/main/java/org/wso2/apk/apimgt/devportal/impl/ApisCommonImpl.java]: removed and resolved

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**